dwave/system/composites/tiling.py

```diff
@@ -104,73 +123,143 @@ def __init__(self, sampler, sub_m, sub_n, t=4):
-
+        #Chimera values (unless pegasus specified)
+        num_sublattices=1
         nodes_per_cell = t * 2
         edges_per_cell = t * t
-        m = n = int(ceil(sqrt(ceil(len(sampler.structure.nodelist) / nodes_per_cell))))  # assume square lattice shape
-        system = dnx.chimera_graph(m, n, t, node_list=sampler.structure.nodelist, edge_list=sampler.structure.edgelist)
-        c2i = {chimera_index: linear_index for (linear_index, chimera_index) in system.nodes(data='chimera_index')}
-        sub_c2i = {chimera_index: linear_index for (linear_index, chimera_index) in tile.nodes(data='chimera_index')}
-
+        if ('topology' in sampler.properties
+            and 'type' in sampler.properties['topology']
+            and 'shape' in sampler.properties['topology']):
+            if sampler.properties['topology']['type'] == 'chimera':
+                #Similar to legacy branch, but using properties information to
+                #robustly define target solver graph (system)
+                if len(sampler.properties['topology']['shape']) != 3: 
+                    raise ValueError('topology shape is not of length 3 '
+                                     '(not compatible with chimera)')
+                if sampler.properties['topology']['shape'][2] != t: 
+                    raise ValueError('Tiling methodology requires that solver'
+                                     'and subproblem have identical shore size')
+                m = sampler.properties['topology']['shape'][0]
+                n = sampler.properties['topology']['shape'][1]
+            else:
+                if len(sampler.properties['topology']['shape']) != 1: 
+                    raise ValueError('topology shape is not of length 1 '
+                                     '(not compatible with pegasus)')
+                #Full yield in odd-couplers also required.
+                #Generalizes chimera subgraph requirement and leads to some 
+                #simplification of expressions, but at with a cost in cell-yield
+                edges_per_cell += t
+                #Square solvers only by pegasus lattice definition PN yields
+                #3 by N-1 by N-1 cells:
+                num_sublattices=3
+                m = n = sampler.properties['topology']['shape'][0] - 1
+                if t!=4:
+                    raise ValueError(
+                        't=4 for all pegasus processors, value is not typically'
+                        'stored in solver properties and is difficult to infer.'
+                        'Therefore only the value t=4 is supported.')
+                    
+        else:
+            warnings.warn("Incomplete solver topology information."
+                          "Falling back to unsafe legacy branch.",
+                          DeprecationWarning)
+            # assume square lattice shape, and high yield, chimera system
+            m = n = int(ceil(sqrt(ceil(len(sampler.structure.nodelist) / nodes_per_cell))))  
```

Comment:
@jackraymond, isn't this calculation off for Pegasus so if the `shape` info is missing from a Pegasus sampler then the `m` here is for Chimera, and using it in line 187 (`system = dnx.pegasus_graph(m, ...`) below will fail?

Reply:
I wanted to keep the original code intact, since it works (except in pathological cases) for Chimera. This deprecation warning is probably not a strong enough disincentive. Better to just throw an error given that Pegasus use case will be increasingly common

Reply:
In other words, deprecate rather than warn.